bundles/org.openhab.core.model.thing/src/org/openhab/core/model/thing/internal/GenericItemChannelLinkProvider.java

```diff
@@ -146,6 +149,17 @@ public void stopConfigurationUpdate(String context) {
             }
         }
         Optional.ofNullable(contextMap.get(context)).ifPresent(ctx -> ctx.removeAll(previousItemNames));
+
+        addedItemChannels.forEach((itemName, addedChannelUIDs) -> {
+            Map<ChannelUID, ItemChannelLink> links = Objects.requireNonNull(itemChannelLinkMap.get(itemName));
```

Comment:
Wouldn't it be more safe to call `itemChannelLinkMap.getOrDefault(itemName, Map.of()`? In case there are no links the current code would result in an NPE.

Reply:
Good idea. I've updated it to use getOrDefault